pre-piksi-multi-launch-updates.md

```diff
@@ -0,0 +1,222 @@
+       - Bits 2-0
+         - 000 Invalid
+         - 001 RTK Float
+         - 010 RTK Fixed
+         - 011 SPP
+         - 100 Reserved
+         - 101 Reserved
+         - 110 Reserved
+         - 111 Reserved
+       - Bits 6-3
+         - Reserved
+       - Bit 7
+         - 0 No RAIM repair
+         - 1 RAIM repair
+       - Note, we remove MSL vs Ellipsoid height as we have nothing but ellipsoid currently
+
+    - BASELINE flags:
+      - Bits: 76543210
+      - Bits 2-0:
+        - 000 Invalid
+        - 001 RTK Float
+        - 010 RTK Fixed
+        - 011 Reserved
+        - 100 Reserved
+        - 101 Reserved
+        - 110 Reserved
+        - 111 Reserved
+      - Bits 6-3:
+        - Reserved
+      - Bit 7:
+        - 0 No RAIM Repair
+        - 1 RAIM repair
+
+     - VEL flags:
+        - Bits: 76543210
+        - Bits 2-0:
+          - 000 Invalid
+          - 001 Measured Doppler Derived
+          - 010 Computed Doppler Derived
+          - 011 Reserved
+          - 100 Reserved
+          - 101 Reserved
+          - 110 Reserved
+          - 111 Reserved
+        - Bits 6-3:
+          - Reserved
+        - Bit 7:
+          - 0 No RAIM Repair
+          - 1 RAIM repair
+
+    - GPS_TIME flags:
+      - Bits: 76543210
+      - Bits 2-0:
+        - 000 Invalid
+        - 001 PVT derived
+        - 010 Reserved (will be filled in when RTC comes online and maybe with coarse time)
+        - 011 Reserved
+        - 100 Reserved
+        - 101 Reserved
+        - 110 Reserved
+        - 111 Reserved
+      - Bits 6-3:
+        - Reserved
+      - Bits 7:
+        - 0 No RAIM repair
+        - 1 RAIM repair
+
+Heading:
+Deprecate for the time being
+
+Dops:
+DOPS calculation needs to update when we have RTK solution or RAIM flag to represent the satellites actually used.  When in RTK mode of any kind, DOPS will represent the DOPS from the RTK solution.  If no RTK solution is available (pseudo absolute or baseline), DOPS will be only SPP DOPS.  Flags will follow the position message definition.
+
+Other option:
+PVT message
+
+### 2. Observation message updates
+
+Updates to observation message to reduce bandwidth and allow observations to
+always be sent regardless of whether they're usable (receiving device filters
+them for usability rather than sending device).  Also allows measured doppler to come over the wire.
+Message has one more byte per observation than current observation message
+We will change max size of observation message to 255 bytes
+
+- Remove week number from header (2 bytes)
+- Replace 4 byte obs.sid with 2 bytes, matching the firmware SID representation
+  - Satellite ID (1 byte)
+    - will be the actual satellite ID value, not value minus 1
+  - Signal (Band/Constellation/Etc) Code (1 byte)
+- Replace 2 byte lock counter with 1 byte lock timer, conforming to DF402 from RTCM 10403.2
+- Add 1 byte of flags for each satellite - need to talk to Estimation and Exafore about what these should be
```

Comment:
Things are still in flux, but I hope we can resolve in https://github.com/swift-nav/libswiftnav-private/pull/245 the exact set of flags this week. We will basically want to support the various NAV_MEAS_* flags from https://github.com/swift-nav/libswiftnav-private/pull/245/files#diff-2ba41cedc176236523f565723eb3901cR533

Some of them we could say are required before generating an SBP observation. E.g. doesn't make sense to send an observation with invalid psuedorange values.